### ISSUE TITLE
Regression of #4198 - Lint/AmbiguousBlockAssociation

### ISSUE THREAD

**Issue by luke-hill** [rubocop/rubocop#7486]
--------

## Expected behavior

`expect { move_direction }.not_to change { current_room_id }` should not raise any offenses
`expect { move_direction }.to change { current_room_id }.to(new_room_id)` should not raise any offenses

## Actual behavior

`expect { move_direction }.not_to change { current_room_id }` raises an offense
`expect { move_direction }.to change { current_room_id }.to(new_room_id)` does not raise an offense

## Steps to reproduce the problem

Code is attached above. I added the working variety to show comparison.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.76.0 (using Parser 2.6.5.0, running on ruby 2.6.3 x86_64-linux)
```

**Comment by mvz** [comment]
I'm still seeing this issue in RuboCop 0.83.0.

**Comment by kreintjes** [comment]
Still an issue in 0.91.0.

**Comment by kreintjes** [comment]
@luke-hill @mvz apparently it is desired behaviour, because the syntax is ambiguous. You can fix it as suggested here: https://github.com/rubocop-hq/rubocop/issues/8732#issuecomment-696913077

In some cases it might also be possible to write the matches as `change(object, :attribute)` instead of using a block, as suggested here: https://github.com/rubocop-hq/rubocop/issues/4198#issuecomment-290335794

Or you can disable the cop in this specific case as suggested by bbatsov here: https://github.com/rubocop-hq/rubocop/issues/4198#issuecomment-290332916

**Comment by luke-hill** [comment]
4198 is impossible here. As I don't have a method on my object output, just the object.

The issue is more than not_to is behaving different to to

**Comment by mvz** [comment]
This seems to be the same behaviour as mentioned in https://github.com/rubocop-hq/rubocop/issues/4198#issuecomment-290338661. I'm not sure if that was actually ever fixed?

As an aside, maybe `change(self, :current_room_id)` works?

**Comment by reedstonefood** [comment]
If you're on rubocop 1.13 or later, then you can use `IgnoreMethods` (as mentioned just above this comment) to deal with this scenario. https://github.com/rubocop/rubocop/pull/9685

**Comment by jimbali** [comment]
Can't you just put parantheses around the change method to make it unambiguous?
```
expect { move_direction }.not_to(change { current_room_id })
```

**Comment by luke-hill** [comment]
Yep that was offered last year by @kreintjes 

If someone from the rubocop technical team says this won't be fix then they can close it. I don't think the original method was that bad?

**Comment by jimbali** [comment]
ah ok, so you're just saying that it's not ambiguous? I was thinking that maybe it was somehow ambiguous in the ruby syntax and that putting parentheses makes it explicitly a method call rather than something else, but I'm not sure what else it would be. It seems like perfectly readable Rspec tbf.

**Comment by reedstonefood** [comment]
Our example of this is similar.

```ruby
expect { subject.perform }.to raise_error { |error|
  # some expectation of error
  # another expectation of error
}
```

This _could_ be fixed as suggested by @jimbali :

```ruby
expect { subject.perform }.to(raise_error { |error|
  # some expectation of error
  # another expectation of error
})
```

However, IMO this reduces readability, which I find to be a fantastic feature of rspec. I don't want extra paranthesis above what is there already. We'll be using the `IgnoreMethods` functionality to stop these being flagged up.

**Comment by bbatsov** [comment]
Can't reproduce as a defect on current master - this is handled via config now: add `change`/`not_change` (etc.) to `Lint/AmbiguousBlockAssociation: AllowedMethods`, available since 1.13. Closing.
